api/src/main/java/com/epam/pipeline/manager/user/UserManager.java

```diff
@@ -95,10 +104,32 @@
     @Autowired
     private MetadataManager metadataManager;
 
-    @Transactional(propagation = Propagation.REQUIRED)
+    @Autowired
+    private FolderManager folderManager;
+
+    @Autowired
+    private GrantPermissionManager permissionManager;
+
+    @Value("${storage.user.home.template}")
+    private  String defaultUserStorageTemplateName;
+
     public PipelineUser createUser(String name, List<Long> roles,
                                    List<String> groups, Map<String, String> attributes,
                                    Long defaultStorageId) {
+        final PipelineUser newUser = createUser(name, roles, groups, attributes);
+        final boolean shouldCreateDefaultHome =
+            preferenceManager.getPreference(SystemPreferences.DEFAULT_USER_DATA_STORAGE_ENABLED);
+        final Long storageId = Optional.ofNullable(defaultStorageId)
```

Comment:
Done, restore transactional behavior for `createUser`, changed methods structure, so `createUser` contains two parts (method calls): user creation itself and default folder initialization. The second one marked as transactional and wrapped into try-catch in `createUser`, so it should revert changes in case of failures, but won't fail a user creation transaction.